src/compiler/tsbuild.ts

```diff
@@ -492,6 +497,27 @@ namespace ts {
             clearMap(allWatchedWildcardDirectories, wildCardWatches => clearMap(wildCardWatches, closeFileWatcherOf));
             clearMap(allWatchedInputFiles, inputFileWatches => clearMap(inputFileWatches, closeFileWatcher));
             clearMap(allWatchedConfigFiles, closeFileWatcher);
+            builderPrograms.clear();
+            updateGetSourceFile();

```

Comment:
why do we have to do this?

Reply:
It seems like it might be related to your comments on EmitAndSemanticDiagnosticsBuilderProgram, but I don't understand how.

Reply:
Currently `resetBuildContext` is API exposed. We are yet to see if that's really needed or is it test only but that api just means clear and be in initial state as if just created.